infra/feast-operator/internal/controller/services/util.go

```diff
@@ -10,70 +12,74 @@ func IsLocalRegistry(featureStore *feastdevv1alpha1.FeatureStore) bool {
+	if services.OfflineStore != nil {
+		setServiceDefaultConfigs(&services.OfflineStore.ServiceConfigs.DefaultConfigs)
+		if services.OfflineStore.Persistence == nil {
+			services.OfflineStore.Persistence = &feastdevv1alpha1.OfflineStorePersistence{}
 		}
-		if applied.Services.OfflineStore.Persistence.FilePersistence == nil {
-			applied.Services.OfflineStore.Persistence.FilePersistence = &feastdevv1alpha1.OfflineStoreFilePersistence{}
+		if services.OfflineStore.Persistence.FilePersistence == nil {
+			services.OfflineStore.Persistence.FilePersistence = &feastdevv1alpha1.OfflineStoreFilePersistence{}
 		}
-		if len(applied.Services.OfflineStore.Persistence.FilePersistence.Type) == 0 {
-			applied.Services.OfflineStore.Persistence.FilePersistence.Type = string(OfflineDaskConfigType)
-		} else {
-			_, err := feastdevv1alpha1.IsValidOfflineStoreFilePersistenceType(applied.Services.OfflineStore.Persistence.FilePersistence.Type)
-			if err != nil {
-				return err
-			}
+		if len(services.OfflineStore.Persistence.FilePersistence.Type) == 0 {
+			services.OfflineStore.Persistence.FilePersistence.Type = string(OfflineDaskConfigType)
 		}
 	}
-	if applied.Services.OnlineStore != nil {
-		setServiceDefaultConfigs(&applied.Services.OnlineStore.ServiceConfigs.DefaultConfigs)
-		if applied.Services.OnlineStore.Persistence == nil {
-			applied.Services.OnlineStore.Persistence = &feastdevv1alpha1.OnlineStorePersistence{}
+	if services.OnlineStore != nil {
+		setServiceDefaultConfigs(&services.OnlineStore.ServiceConfigs.DefaultConfigs)
+		if services.OnlineStore.Persistence == nil {
+			services.OnlineStore.Persistence = &feastdevv1alpha1.OnlineStorePersistence{}
 		}
-		if applied.Services.OnlineStore.Persistence.FilePersistence == nil {
-			applied.Services.OnlineStore.Persistence.FilePersistence = &feastdevv1alpha1.OnlineStoreFilePersistence{}
+		if services.OnlineStore.Persistence.FilePersistence == nil {
+			services.OnlineStore.Persistence.FilePersistence = &feastdevv1alpha1.OnlineStoreFilePersistence{}
 		}
-		if len(applied.Services.OnlineStore.Persistence.FilePersistence.Path) == 0 {
-			applied.Services.OnlineStore.Persistence.FilePersistence.Path = DefaultOnlinePath
+		if len(services.OnlineStore.Persistence.FilePersistence.Path) == 0 {
+			services.OnlineStore.Persistence.FilePersistence.Path = DefaultOnlinePath
 		}
 	}
 	// overwrite status.applied with every reconcile
 	applied.DeepCopyInto(&cr.Status.Applied)
-	return nil
 }
 
 func setServiceDefaultConfigs(defaultConfigs *feastdevv1alpha1.DefaultConfigs) {
 	if defaultConfigs.Image == nil {
 		defaultConfigs.Image = &DefaultImage
 	}
 }
+
+func isValidOfflineStoreFilePersistenceType(value string) (bool, error) {
+	for _, v := range feastdevv1alpha1.ValidOfflineStoreFilePersistenceTypes {
+		if v == value {
+			return true, nil
+		}
+	}
+	return false, fmt.Errorf("invalid  file type %s for offline store", value)
+}
```

Comment:
i appears the `bool` return isn't being used anywhere. we should probably either remove it or use it instead of the `error` return. there doesn't currently appear to be a need for both. thoughts?

Reply:
probably makes more sense keep the `error` and remove the `bool`. this would necessitate that the function named changed to something like `checkOfflineStoreFilePersistenceType` instead.